### PR TITLE
Show correct attribute values in Indirect Data Analysis FunctionBrowser

### LINKED ISSUE #28529: Make Q values update correctly in FunctionBrowser
Currently Q values do not update in the function browser when the dataset being observed is changed. This is due to a the FunctionBrowser not handling attritubute updating. This needs to be resolved so the Q values used are resolved correctly.

### PR DESCRIPTION
**Pull request by StephenSmith25**
**Description of work.**
This PR adds a missing functions to manipulate attributes in `IFunctionModel` and `IFunctionView`. Both classes were missing `getAttribute` and `setAttribute` methods. Due to changes made in a previous PR, these can now be dealt with in the exact same manner as `setParameter(..)`.

The primary motivation for this work was to show the correct `Q` value for `Q` dependent functions in the `ConvFit` tab of the indirect data analysis GUI, as such as the testing instructions will relate to this issue:

**Test instructions**

1. Open the Indirect data analysis GUI
2. Go to Conv fit and load data from #26631
3. Add the fit type as the TeixeiraWater function
4. Change to full function view, you should note that the correct `Q` value is shown
5. Switch between spectra and the Q value will change
6. Change the `Q` value and switch on/off the full function view, you should note that the `Q` value that you set remained – signifying that the model has been appropriately updated.

As an aside, `Q` values set in this browser are not used in the Convolution fit performed in this Tab, as the 'Q' values are fetched from the workspace on-the-fly when a fit is requested (which explains why the correct values were used before even when the browser showed the incorrect Q values). Fixing this issue (if desired) would require another PR.

Fixes #28529 . 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by Harrietbrown — CHANGES_REQUESTED** [review]

File: qt/widgets/common/src/FunctionMultiDomainPresenter.cpp [hunk 1]
```diff
@@ -25,6 +25,37 @@ namespace MantidWidgets {
...
+/**
+ * Attribute visitor to set an attribute in an IFunctionView.
+ */
+class SetAttributeInView : public IFunction::AttributeVisitor<> {
+public:
+  SetAttributeInView(IFunctionView *view, QString attrName)
+      : m_view(view), m_attrName(attrName){};
+
+protected:
+  void apply(double &d) const override {
```
[the remark this answers is not in the document]
Comment (StephenSmith25) [reply qt/widgets/common/src/FunctionMultiDomainPresenter.cpp:37]: Annoyingly these override a base class method so they cannot be templated. I’ll have a look at doing something different, as it is just writing the same line over and over again.. Doesn’t seem great.

[the remark this answers is not in the document]
Comment (Harrietbrown) [reply qt/widgets/common/src/FunctionMultiDomainPresenter.cpp]: how much work would it be to change the base class method? do the other implementations of it have any differences that would stop them from being templated?

[the remark this answers is not in the document]
Comment (StephenSmith25) [reply qt/widgets/common/src/FunctionMultiDomainPresenter.cpp]: Latest commit should remove some boilerplate and allow lambda expression to be used. For our use case this means we can simple pass in a single lambda, without having to define that `struct` at the top of the .cpp file.

**Review by Harrietbrown — APPROVED** [review]
The Q values are corectly set and remembers when chaged in the browser. Good Job

**Pull request merged**